pvnet/models/multimodal/validation/config_batch_validation.py

```diff
@@ -78,74 +78,88 @@ def get_modality_interval(
             positive integer), or if `secondary_modality_key` is required
             for NWP but not provided.
     """
-    config_to_check = input_data_config
-    lookup_key = primary_modality_key
-    error_context = f"modality '{primary_modality_key}'"
-
-    if is_nwp_source:
-        if primary_modality_key not in config_to_check:
-            raise KeyError(f"NWP section ('{primary_modality_key}') not found in input_data_config")
-        config_to_check = config_to_check[primary_modality_key]
-        if not isinstance(config_to_check, dict):
-            raise TypeError(f"Expected dict for '{primary_modality_key}' in input_data_config")
+    config_section_to_search = input_data_config
+    key_for_config = primary_modality_key
+    error_context_desc = f"modality '{primary_modality_key}'"
+
+    if primary_modality_key == "nwp":
         if secondary_modality_key is None:
             raise ValueError(
-                "secondary_modality_key (NWP source) is required when is_nwp_source=True"
+                "When primary_modality_key is 'nwp', secondary_modality_key "
+                "(the NWP source name) must be provided."
             )
-
-        lookup_key = secondary_modality_key
-        error_context = f"NWP source '{lookup_key}'"
-
-    modality_config_dict = None
-    if lookup_key not in config_to_check:
-        if (primary_modality_key == "sun"
-                and secondary_modality_key is not None
-                and secondary_modality_key in input_data_config):
-            logger.debug(f"Using '{secondary_modality_key}' interval as fallback for sun.")
-            lookup_key = secondary_modality_key
-            error_context = f"fallback modality '{lookup_key}' for sun"
-            modality_config_dict = input_data_config.get(lookup_key)
+        if "nwp" not in input_data_config or not isinstance(input_data_config["nwp"], dict):
+                raise KeyError(
+                    "NWP section ('nwp') not found or is not a dictionary "
+                    "in input_data_config."
```

Comment:
I'm continuing to split hairs but I think this is clear:
```suggestion
                    "in data configuration."
```
because input_data_config is an internal variable here and the actual config can be called something completely different